fdbserver/ptxn/test/RealStorageServer.actor.cpp

```diff
@@ -0,0 +1,180 @@
+	ActorCollection actors = ActorCollection(false);
+
+	StorageServerTestDriver() : ServerTestDriver() { initEndpoints(); }
+
+	void initEndpoints() {
+		UID uid = nondeterministicRandom()->randomUniqueID();
+		ssi = StorageServerInterface(uid);
+		ssi.locality = LocalityData();
+		ssi.initEndpoints();
+	}
+
+	ACTOR static Future<Void> runStorageServer(StorageServerTestDriver* self) {
+		wait(delay(1));
+
+		StorageServerInterface& ssi = self->ssi;
+
+		KeyValueStoreType storeType = KeyValueStoreType::SSD_BTREE_V2;
+		state std::string folder = ".";
+		std::string fileName = joinPath(folder, "storage-" + ssi.id().toString() + "." + storeType.toString());
+		std::cout << "new Storage Server file name: " << fileName << std::endl;
+		deleteFile(fileName);
+		state IKeyValueStore* data = openKVStore(storeType, fileName, ssi.id(), 0);
+
+		state ReplyPromise<InitializeStorageReply> storageReady;
+
+		// Initialize dbInfo with mocked cursor and mocked loy system.
+
+		// TODO: Use Reference API?
+		// TODO: delete mockLogSystem eventually.
+		state MockLogSystem* mockLogSystem = new MockLogSystem();
+		mockLogSystem->cursor = self->mockPeekCursor.castTo<ILogSystem::IPeekCursor>();
+
+		LogSystemConfig logSystemConfig;
+		logSystemConfig.logSystemType = LogSystemType::mock;
+		logSystemConfig.mockLogSystem = mockLogSystem;
+
+		ServerDBInfo dbInfoBuilder;
+		dbInfoBuilder.recoveryState = RecoveryState::ACCEPTING_COMMITS;
+		dbInfoBuilder.logSystemConfig = logSystemConfig;
+		state Reference<AsyncVar<ServerDBInfo>> dbInfo = makeReference<AsyncVar<ServerDBInfo>>(dbInfoBuilder);
+		// Initialize dbInfo done.
+
+		std::cout << "Starting Storage Server." << std::endl;
+		state Future<Void> ss = storageServer(data, ssi, self->tag, storageReady, dbInfo, folder);
+		std::cout << "Storage Server started." << std::endl;
+
+		self->actors.add(ss);
+		wait(ss);
+
+		return Void();
+	}
+
+	ACTOR static Future<Void> verifyGetValue(StorageServerTestDriver* self,
+	                                         KeyRef key,
+	                                         Version version,
+	                                         ValueRef expectedValue) {
+		wait(switchToServerProcess(self));
+
+		GetValueRequest getValueRequest = GetValueRequest(UID(), // spanContext
+		                                                  key,
+		                                                  version,
+		                                                  Optional<TagSet>(), // tags
+		                                                  Optional<UID>()); // debugID
+		self->ssi.getValue.send(getValueRequest);
+
+		GetValueReply getValueReply = wait(getValueRequest.reply.getFuture());
+		const Value& value = getValueReply.value.get();
+		std::cout << "Get value: " << value.toString() << std::endl;
+		ASSERT(value == expectedValue);
+
+		wait(switchBack(self));
+		return Void();
+	}
+};
+} // namespace ptxn
+
+TEST_CASE("fdbserver/ptxn/test/storageserver") {
+	using namespace ptxn;
```

Comment:
AFAIU, there is no FDB style guide rule prohibiting `using namespace`, but it is used rarely for good reason. The risk of conflicting symbol names is fairly high and can cause a fair bit of maintenance pain.